src/parse_sync.cc

```diff
@@ -42,22 +40,30 @@ NAN_METHOD(protagonist::ParseSync) {
             Nan::ThrowTypeError(optionsResult->error);
         }
 
-        options = optionsResult->options;
-        astType = optionsResult->astType;
+        parseOptions = optionsResult->parseOptions;
+        serializeOptions = optionsResult->serializeOptions;
         FreeOptionsResult(&optionsResult);
     }
 
     // Parse the source data
-    snowcrash::ParseResult<snowcrash::Blueprint> parseResult;
-    snowcrash::parse(*sourceData, options | snowcrash::ExportSourcemapOption, parseResult);
-
-    Local<Object> result = v8_wrap(Result::WrapResult(parseResult, options, astType))->ToObject();
-
-    // Error Object (FIXME: Don't have this once we remove AST)
-    if (parseResult.report.error.code != snowcrash::Error::OK) {
-        Nan::ThrowError((Local<Value>) SourceAnnotation::WrapSourceAnnotation(parseResult.report.error));
-        return;
+    drafter_result *result;
+    int err_code = drafter_parse_blueprint(*sourceData,
+                                           &result,
+                                           parseOptions);
+    switch (err_code) {
+    case DRAFTER_EUNKNOWN:
+        Nan::ThrowError("Parser: Unknown Error");
+        break;
+    case DRAFTER_EINVALID_INPUT:
+        Nan::ThrowError("Parser: Invalid Input");
```

Comment:
> Just have a proper sentence here instead of just saying "Parser: Invalid Input". For example, "The given input is not valid. Please check again."

I am not sure that this is much better. I think there are 3 points that are good to cover in a good error message:

- The problem
- The cause
- The solution

---

Looking through the drafter source code to understand what the underlying problem is. In this case it is when the `source` is `NULL`. This shouldn't be possible in Protagonist since we are always creating the source: https://github.com/apiaryio/protagonist/pull/189/files#diff-7422bf7954365fe2adea00e4d17f87c4R89.

If a user of Protagonist was to receive this error, I am not sure there is anything they can do as it would likely be a regression we introduced.

I think in the case of `DRAFTER_EINVALID_OUTPUT` is is more definitive, if `DRAFTER_EINVALID_OUTPUT` would happen then Protagonist didn't supply an output buffer which would be a programming error on our side.

That leaves `DRAFTER_EUNKNOWN`, and after looking at the source code this won't happen so there is no cause. It seems to be a placeholder for future unknown errors.